### ISSUE TITLE
Ability to create or specify Cache Subnet Group in ElastiCache module

### ISSUE THREAD

**Issue by tiyberius** [ansible/ansible-modules-core#1089]
##### Issue Type:

Bug Report
##### Ansible Version:

$ ansible --version
ansible 1.9.0.1
##### Environment:

Mac OS 10.10.2
##### Summary:

I'm trying to create a VPC, a security group, and an ElastiCache cluster. The cluster should be created in the VPC and when it is created, should be assigned to a VPC security group.
##### Steps To Reproduce:

After #1088 is resolved, run the following playbook

``` ---
- name: Demonstrate Ansible ElastiCache bug
  hosts: localhost
  connection: local
  gather_facts: False
  tasks:

  - name: Create a VPC
    local_action:
      module: ec2_vpc
      state: present
      cidr_block: 10.0.0.0/16
      resource_tags:
        Name: Test VPC please delete
      subnets:
      - cidr: 10.0.1.0/24
        az: us-east-1a
      - cidr: 10.0.2.0/24
        az: us-east-1c
      route_tables:
        - subnets:
          - 10.0.1.0/24
          routes:
            - dest: 0.0.0.0/0
              gw: igw
      internet_gateway: True
      region: us-east-1
      wait: true
    register: our_vpc
    until: vpc
    retries: 2
    delay: 5

  - name: Create the security group for Redis in ElastiCache
    local_action:
      module: ec2_group
      name: Redis Security Group
      description: Created for demonstration of elasticache module bug
      vpc_id: "{{ our_vpc.vpc_id }}"
      region: us-east-1
      rules: []
    ignore_errors: yes
    register: redis_security_group

  - name: Create a Redis cluster in ElastiCache
    elasticache:
      name: redis-cluster
      state: present
      engine: redis
      cache_engine_version: 2.8.6
      node_type: cache.r3.large
      num_nodes: 1
      security_group_ids: "{{ redis_security_group.group_id }}"
      region: us-east-1
      zone: us-east-1a
```
##### Expected Results:

I expected a Redis ElastiCache cluster inside a VPC that is assigned to a security group.
##### Actual Results:

```
PLAY [Demonstrate Ansible ElastiCache bug] ************************************

TASK: [Create a VPC] **********************************************************
changed: [localhost -> 127.0.0.1]

TASK: [Create the security group for Redis in ElastiCache] ********************
ok: [localhost -> 127.0.0.1]

TASK: [Create a Redis cluster in ElastiCache] *********************************
failed: [localhost] => {"failed": true}
msg: {"Error":{"Code":"InvalidParameterCombination","Message":"Subnet group [default] belongs to a different VPC [vpc-********] than [vpc-********]","Type":"Sender"},"RequestId":"7b220059-dbca-11e4-8618-edd76ff04598"}

FATAL: all hosts have already failed -- aborting

PLAY RECAP ********************************************************************
           to retry, use: --limit @/Users/twork/elasticache_bug.retry

localhost                  : ok=2    changed=1    unreachable=0    failed=1
```

**Comment by bennojoy** [comment]
@tiyberius probably what you need would be to try this PR https://github.com/ansible/ansible-modules-core/pull/1137 to create a cache_subnet_group with the subnet_ids of the vpc that you created above and then pass that subnet group name to ' cache_subnet_group' parameter and things should work fine, can you please give it a try..

**Comment by worldofchris** [comment]
@tiyberius +1 to what @bennojoy said.  I just used PR #1137 and now have an Elasticache node running in my VPC.

**Comment by tiyberius** [comment]
@bennojoy sorry just seeing your comment now, not sure how I missed it when you commented 2 weeks ago. I just realized I have no idea how to test #1137 and I'm tight on time today, so I will trust that it works! Thanks for the fix

**Comment by bcoca** [comment]
closing via merged #1137
